ratis-common/src/main/java/org/apache/ratis/protocol/RoutingTable.java

```diff
@@ -159,7 +164,7 @@ public String toString() {
 
   /** @return a new {@link RoutingTable} represented by the given map. */
   static RoutingTable newRoutingTable(Map<RaftPeerId, Set<RaftPeerId>> map){
-    Builder.validate(map);
+    RaftPeerId primary = Builder.validate(map);
```

Comment:
Let's return null when `map == null || map.isEmpty()`.  Then, we can assume the map is non-empty from here.

Reply:
Thank you for the review. Updated based on the patch diff.

Reply:
It seems this will trigger the `IllegalStateException` in `RoutingTable#Builder#build` due to `newRoutingTable` returning `null`.  

See failed CI: https://github.com/ivandika3/ratis/actions/runs/6602996134/job/17935643444

Should I revert this and also re-add the check that `routingTable.getPrimary() != null` ?

Reply:
Let's update the build() method.
```java
//RoutingTable
    public RoutingTable build() {
      final Map<RaftPeerId, Set<RaftPeerId>> map = ref.getAndSet(null);
      if (map == null) {
        throw new IllegalStateException("RoutingTable Already built");
      }
      return RoutingTable.newRoutingTable(map);
    }
```

Reply:
Thank you for the suggestion. Updated accordingly.